### ISSUE TITLE
Testsuite failures on PHP7

### ISSUE THREAD

**Issue by shanethehat** [MageTest/MageSpec#128]
It seems that changes to the typehinting system in PHP7 have broken controller generation, as illustrated by the failures in the testsuite:

```
shane@jeeves magespec (develop)*$ bin/behat features/class_generation.feature:12 -v
Feature: Magento classes are generated from specs
  So that I can practice SpecBDD with Magento
  As a developer
  I want Magespec to generate Magento classes from specs

  Scenario: Controller class is generated from a spec # features/class_generation.feature:12
    Given Magespec has a standard configuration       # FeatureContext::magespecHasAStandardConfiguration()
    And there is a "controller" spec                  # FeatureContext::thatThereIsASpec()
    When Magespec runs the spec                       # FeatureContext::magespecRunsTheSpec()
      Behat\Testwork\Call\Exception\FatalThrowableError: Type error: Argument 1 passed to MageTest\PhpSpec\MagentoExtension\Specification\ControllerBehavior::let() must be an instance of Mage_Core_Controller_Request_Http, instance of PhpSpec\Wrapper\Collaborator given in src/MageTest/PhpSpec/MagentoExtension/Specification/ControllerBehavior.php:40
      Stack trace:
      #0 [internal function]: MageTest\PhpSpec\MagentoExtension\Specification\ControllerBehavior->let(Object(PhpSpec\Wrapper\Collaborator), Object(PhpSpec\Wrapper\Collaborator))
      #1 vendor/phpspec/phpspec/src/PhpSpec/Runner/Maintainer/LetAndLetgoMaintainer.php(52): ReflectionMethod->invokeArgs(Object(spec\Behat_Spec_TestControllerSpec), Array)
      #2 vendor/phpspec/phpspec/src/PhpSpec/Runner/ExampleRunner.php(136): PhpSpec\Runner\Maintainer\LetAndLetgoMaintainer->prepare(Object(PhpSpec\Loader\Node\ExampleNode), Object(spec\Behat_Spec_TestControllerSpec), Object(PhpSpec\Runner\MatcherManager), Object(PhpSpec\Runner\CollaboratorManager))
      #3 vendor/phpspec/phpspec/src/PhpSpec/Runner/ExampleRunner.php(79): PhpSpec\Runner\ExampleRunner->executeExample(Object(spec\Behat_Spec_TestControllerSpec), Object(PhpSpec\Loader\Node\ExampleNode))
      #4 vendor/phpspec/phpspec/src/PhpSpec/Runner/SpecificationRunner.php(55): PhpSpec\Runner\ExampleRunner->run(Object(PhpSpec\Loader\Node\ExampleNode))
      #5 vendor/phpspec/phpspec/src/PhpSpec/Runner/SuiteRunner.php(56): PhpSpec\Runner\SpecificationRunner->run(Object(PhpSpec\Loader\Node\SpecificationNode))
      #6 vendor/phpspec/phpspec/src/PhpSpec/Console/Command/RunCommand.php(172): PhpSpec\Runner\SuiteRunner->run(Object(PhpSpec\Loader\Suite))
      #7 vendor/symfony/console/Command/Command.php(256): PhpSpec\Console\Command\RunCommand->execute(Object(Symfony\Component\Console\Input\ArrayInput), Object(Symfony\Component\Console\Output\StreamOutput))
      #8 vendor/symfony/console/Application.php(807): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArrayInput), Object(Symfony\Component\Console\Output\StreamOutput))
      #9 vendor/symfony/console/Application.php(186): Symfony\Component\Console\Application->doRunCommand(Object(PhpSpec\Console\Command\RunCommand), Object(Symfony\Component\Console\Input\ArrayInput), Object(Symfony\Component\Console\Output\StreamOutput))
      #10 vendor/phpspec/phpspec/src/PhpSpec/Console/Application.php(97): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArrayInput), Object(Symfony\Component\Console\Output\StreamOutput))
      #11 vendor/symfony/console/Application.php(117): PhpSpec\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArrayInput), Object(Symfony\Component\Console\Output\StreamOutput))
      #12 vendor/symfony/console/Tester/ApplicationTester.php(76): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArrayInput), Object(Symfony\Component\Console\Output\StreamOutput))
      #13 features/bootstrap/FeatureContext.php(227): Symfony\Component\Console\Tester\ApplicationTester->run(Array, Array)
      #14 [internal function]: FeatureContext->magespecRunsTheSpec()
      #15 vendor/behat/behat/src/Behat/Testwork/Call/Handler/RuntimeCallHandler.php(104): call_user_func_array(Array, Array)
      #16 vendor/behat/behat/src/Behat/Testwork/Call/Handler/RuntimeCallHandler.php(60): Behat\Testwork\Call\Handler\RuntimeCallHandler->executeCall(Object(Behat\Behat\Definition\Call\DefinitionCall))
      #17 vendor/behat/behat/src/Behat/Testwork/Call/CallCenter.php(129): Behat\Testwork\Call\Handler\RuntimeCallHandler->handleCall(Object(Behat\Behat\Definition\Call\DefinitionCall))
      #18 vendor/behat/behat/src/Behat/Testwork/Call/CallCenter.php(82): Behat\Testwork\Call\CallCenter->handleCall(Object(Behat\Behat\Definition\Call\DefinitionCall))
      #19 vendor/behat/behat/src/Behat/Behat/Tester/Runtime/RuntimeStepTester.php(125): Behat\Testwork\Call\CallCenter->makeCall(Object(Behat\Behat\Definition\Call\DefinitionCall))
      #20 vendor/behat/behat/src/Behat/Behat/Tester/Runtime/RuntimeStepTester.php(73): Behat\Behat\Tester\Runtime\RuntimeStepTester->testDefinition(Object(Behat\Behat\Context\Environment\InitializedContextEnvironment), Object(Behat\Gherkin\Node\FeatureNode), Object(Behat\Gherkin\Node\StepNode), Object(Behat\Behat\Definition\SearchResult), false)
      #21 vendor/behat/behat/src/Behat/Behat/Hook/Tester/HookableStepTester.php(74): Behat\Behat\Tester\Runtime\RuntimeStepTester->test(Object(Behat\Behat\Context\Environment\InitializedContextEnvironment), Object(Behat\Gherkin\Node\FeatureNode), Object(Behat\Gherkin\Node\StepNode), false)
      #22 vendor/behat/behat/src/Behat/Behat/EventDispatcher/Tester/EventDispatchingStepTester.php(73): Behat\Behat\Hook\Tester\HookableStepTester->test(Object(Behat\Behat\Context\E  1 {
nvironment\InitializedContextEnvironment), Object(Behat\Gherkin\Node\FeatureNode), Object(Behat\Gherkin\Node\StepNode), false)
      #23 vendor/behat/behat/src/Behat/Behat/Tester/StepContainerTester.php(59): Behat\Behat\EventDispatcher\Tester\EventDispatchingStepTester->test(Object(Behat\Behat\Context\Environment\InitializedContextEnvironment), Object(Behat\Gherkin\Node\FeatureNode), Object(Behat\Gherkin\Node\StepNode), false)
      #24 vendor/behat/behat/src/Behat/Behat/Tester/Runtime/RuntimeScenarioTester.php(76): Behat\Behat\Tester\StepContainerTester->test(Object(Behat\Behat\Context\Environment\InitializedContextEnvironment), Object(Behat\Gherkin\Node\FeatureNode), Object(Behat\Gherkin\Node\ScenarioNode), false)
      #25 vendor/behat/behat/src/Behat/Behat/Hook/Tester/HookableScenarioTester.php(74): Behat\Behat\Tester\Runtime\RuntimeScenarioTester->test(Object(Behat\Behat\Context\Environment\InitializedContextEnvironment), Object(Behat\Gherkin\Node\FeatureNode), Object(Behat\Gherkin\Node\ScenarioNode), false)
      #26 vendor/behat/behat/src/Behat/Behat/EventDispatcher/Tester/EventDispatchingScenarioTester.php(103): Behat\Behat\Hook\Tester\HookableScenarioTester->test(Object(Behat\Behat\Context\Environment\InitializedContextEnvironment), Object(Behat\Gherkin\Node\FeatureNode), Object(Behat\Gherkin\Node\ScenarioNode), false)
      #27 vendor/behat/behat/src/Behat/Behat/Tester/Runtime/RuntimeFeatureTester.php(83): Behat\Behat\EventDispatcher\Tester\EventDispatchingScenarioTester->test(Object(Behat\Behat\Context\Environment\InitializedContextEnvironment), Object(Behat\Gherkin\Node\FeatureNode), Object(Behat\Gherkin\Node\ScenarioNode), false)
      #28 vendor/behat/behat/src/Behat/Behat/Hook/Tester/HookableFeatureTester.php(72): Behat\Behat\Tester\Runtime\RuntimeFeatureTester->test(Object(Behat\Behat\Context\Environment\UninitializedContextEnvironment), Object(Behat\Gherkin\Node\FeatureNode), false)
      #29 vendor/behat/behat/src/Behat/Behat/EventDispatcher/Tester/EventDispatchingFeatureTester.php(71): Behat\Behat\Hook\Tester\HookableFeatureTester->test(Object(Behat\Behat\Context\Environment\UninitializedContextEnvironment), Object(Behat\Gherkin\Node\FeatureNode), false)
      #30 vendor/behat/behat/src/Behat/Testwork/Tester/Runtime/RuntimeSuiteTester.php(63): Behat\Behat\EventDispatcher\Tester\EventDispatchingFeatureTester->test(Object(Behat\Behat\Context\Environment\UninitializedContextEnvironment), Object(Behat\Gherkin\Node\FeatureNode), false)
      #31 vendor/behat/behat/src/Behat/Testwork/Hook/Tester/HookableSuiteTester.php(73): Behat\Testwork\Tester\Runtime\RuntimeSuiteTester->test(Object(Behat\Behat\Context\Environment\UninitializedContextEnvironment), Object(Behat\Testwork\Specification\GroupedSpecificationIterator), false)
      #32 vendor/behat/behat/src/Behat/Testwork/EventDispatcher/Tester/EventDispatchingSuiteTester.php(72): Behat\Testwork\Hook\Tester\HookableSuiteTester->test(Object(Behat\Behat\Context\Environment\UninitializedContextEnvironment), Object(Behat\Testwork\Specification\GroupedSpecificationIterator), false)
      #33 vendor/behat/behat/src/Behat/Testwork/Tester/Runtime/RuntimeExercise.php(71): Behat\Testwork\EventDispatcher\Tester\EventDispatchingSuiteTester->test(Object(Behat\Behat\Context\Environment\UninitializedContextEnvironment), Object(Behat\Testwork\Specification\GroupedSpecificationIterator), false)
      #34 vendor/behat/behat/src/Behat/Testwork/EventDispatcher/Tester/EventDispatchingExercise.php(70): Behat\Testwork\Tester\Runtime\RuntimeExercise->test(Array, false)
      #35 vendor/behat/behat/src/Behat/Testwork/Ordering/OrderedExercise.php(80): Behat\Testwork\EventDispatcher\Tester\EventDispatchingExercise->test(Array, false)
      #36 vendor/behat/behat/src/Behat/Testwork/Tester/Cli/ExerciseController.php(146): Behat\Testwork\Ordering\OrderedExercise->test(Array, false)
      #37 vendor/behat/behat/src/Behat/Testwork/Tester/Cli/ExerciseController.php(108): Behat\Testwork\Tester\Cli\ExerciseController->testSpecifications(Object(Symfony\Component\Console\Input\ArgvInput), Array)
      #38 vendor/behat/behat/src/Behat/Testwork/Cli/Command.php(63): Behat\Testwork\Tester\Cli\ExerciseController->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      #39 vendor/symfony/console/Command/Command.php(256): Behat\Testwork\Cli\Command->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      #40 vendor/symfony/console/Application.php(791): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      #41 vendor/symfony/console/Application.php(186): Symfony\Component\Console\Application->doRunCommand(Object(Behat\Testwork\Cli\Command), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      #42 vendor/behat/behat/src/Behat/Testwork/Cli/Application.php(121): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      #43 vendor/symfony/console/Application.php(117): Behat\Testwork\Cli\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
      #44 vendor/behat/behat/bin/behat(32): Symfony\Component\Console\Application->run()
      #45 {main}
    Then a controller class should be generated       # FeatureContext::aControllerClassShouldBeGenerated()

--- Failed scenarios:

    features/class_generation.feature:12

1 scenario (1 failed)
4 steps (2 passed, 1 failed, 1 skipped)
0m0.46s (10.64Mb)
```

**Comment by maxbaldanza** [comment]
Fixed those errors here: https://github.com/maxbaldanza/MageSpec/compare/make-compatible-with-phpspec-3.3...maxbaldanza:make-compatible-with-php7?expand=1

Just waiting on the following PR to be merged before opening a new PR: https://github.com/MageTest/MageSpec/pull/140

**Comment by maxbaldanza** [comment]
PR to fix this here: https://github.com/MageTest/MageSpec/pull/141

**Comment by maxbaldanza** [comment]
https://github.com/MageTest/MageSpec/pull/141 has now been merged so closing this 😄
